app/controllers/recryptrequests_controller.rb

```diff
@@ -44,20 +71,26 @@ def create_recrypt_request
     redirect_to session_destroy_path
   end
 
-  def recrypt_private_key
-    if current_user.recrypt_private_key!(params[:new_password], params[:old_password])
+  def recrypt_private_key(new_password: nil)
+    new_password ||= params[:new_password]
+    if current_user.recrypt_private_key!(new_password, params[:old_password])
+      update_provider
       flash[:notice] = t('flashes.recryptrequests.recrypted')
-      return redirect_to session_destroy_path
+      return redirect_to sso_path
```

Comment:
Can you just redirect to the sso_path here? Doesn't that conflict with other authenticators? We do still use it inside the recrypt method